### DISCUSSION TITLE
How to configure an instance of AnnotatedControllerConfigurer

### DISCUSSION THREAD

**Discussion by henry-filosa** [graphql-java/graphql-java#3055] [Q&A] [answered]
I'm trying to define a configuration to create an `ExecutionGraphQlService` to wire into an existing application as a proof of concept. Here is my attempted code:
```
@Configuration
public class GraphQlConfig {

    @Bean
    public ExecutionGraphQlService defaultExecutionGraphQlService(){
        GraphQlSource.SchemaResourceBuilder schemaBuilder = GraphQlSource.schemaResourceBuilder();
        PathMatchingResourcePatternResolver pathmatcher = new PathMatchingResourcePatternResolver();
        try {
            schemaBuilder.schemaResources(pathmatcher.getResources("graphql/*"));
        } catch (IOException e) {
            e.printStackTrace();
        }
        RuntimeWiring.Builder runtimeWiringBuilder = RuntimeWiring.newRuntimeWiring();

        // Here's the issue
        AnnotatedControllerConfigurer annotatedControllerConfigurer = new AnnotatedControllerConfigurer();
        annotatedControllerConfigurer.configure(runtimeWiringBuilder);

        schemaBuilder.configureRuntimeWiring(annotatedControllerConfigurer);
        GraphQlSource graphQlSource = schemaBuilder.build();
        return new DefaultExecutionGraphQlService(graphQlSource);
    }
}
```
My understanding is that `GraphQlSource.SchemaResourceBuilder` expects `schemaResources` which us used to locates Graph ql schema files and a runtime wiring which maps data fields in the schema to methods that return those fields. I believe I have the first part working, but I've hit a dead end with the second.
I can't get a properly configured `AnnotatedControllerConfigurer` instance, which I would like as it seems to handle the scanning of my controller beans for annotated methods to map to field values. `AnnotatedControllerConfigurer` implements `ApplicationContextAware, InitializingBean` so it seems to expect to be initialized as a bean, but attempts to autowire an instance of it fail due to `Could not autowire. No beans of 'AnnotatedControllerConfigurer' type found. `.

I've attempted constructor injection `public ExecutionGraphQlService defaultExecutionGraphQlService(AnnotatedControllerConfigurer annotatedControllerConfigurer)` as well as manual instantiation `autowireCapableBeanFactory.autowireBean(annotatedControllerConfigurer)` (which should be basically the same thing). 
It seems like this class is not a bean, but expects to be managed by Spring's bean lifecycle?

How is this instance supposed to be created?

The docs are [frustratingly opaque on this](https://docs.spring.io/spring-graphql/docs/current/reference/html/#controllers-declaration) and I have looked in the Spring boot implementation referenced, but could find not mentions of `AnnotatedControllerConfigurer`.

Any guidance appreciated.

**Comment by dondonz** [comment]
Hello, could you also cross post this on the Spring for GraphQL repo? This repo is for the GraphQL Java engine, the controller code would be inside the Spring for GraphQL project https://github.com/spring-projects/spring-graphql/issues

**Comment by bclozel** [comment, marked as answer]
@henry-filosa You can raise this question on [StackOverflow with the spring-graphql](https://stackoverflow.com/questions/tagged/spring-graphql) tag, we'll get back to you there. You can also [create a new issue](https://github.com/spring-projects/spring-graphql/issues) asking for a documentation enhancement.

**Reply by henry-filosa** [reply]
[Created a post here](https://stackoverflow.com/questions/74827708/how-to-properly-create-an-instance-of-annotatedcontrollerconfigurer)

**Comment by henry-filosa** [comment]
Sorry for the mispost! I got turned around looking for the right forum. Will post there as suggested, thanks!
